### DISCUSSION TITLE
Migrating NextAuth User To Supabase

### DISCUSSION THREAD

**Discussion by mwolf1989** [supabase/supabase#21886] [Questions]
Hello everyone,

i have a project with already around 1000 paying users in a mongodb with nextauth and stripe, all users are email/password auth.

is there a possibility to migrate the users and use supabase auth and keep their passwords? i would not like to work with nextauth and the supabase adapter.

Thanks in advance,

Marcel

**Comment by nischalj10** [comment]
hey @mwolf1989 - i am in a similar situation. could you figure out how to do this? thanks!

**Reply by mwolf1989** [reply]
I did. I tried to Write a Blog article ( I am not a good writer) 

https://marcelwolf.dev/blog/supabase-create-user/

let me know is something is unclear

**Reply by GadMugambi** [reply]
Hello. 

The blog is not available. 

I'm facing a similar issue. 

Does supabase provide tooling to migrate from Kinde Auth to Supabase Auth? 

I have around a thousand users 

The main issue is the user ids from Kinde, which I've used extensively in my database. 

Is it possible to alter the user ids created by supabase auth and replace them with the ids from Kinde?

Which is the best way to go about this?

**Reply by mwolf1989** [reply]
Hi the article moved here:

https://marcelwolf.dev/blog/migrate-user-auth-supabase/

**Reply by GadMugambi** [reply]
Thanks a lot.

The issue is now retaining the user ids from Kinde, as I've used them extensively in my database.

How did you go about that?

Were you using uuids when using Auth.js?

**Reply by mwolf1989** [reply]
I did not rely on uuids but on the email address. 

if you check the function, I generate uuids on the fly:

DECLARE
  user_id uuid;
  encrypted_pw text;
  app_metadata jsonb;
  confirmation timestamp;
  firstname text := user_meta_data->>'firstName';
  lastname text := user_meta_data->>'lastName';
BEGIN
  -- set variables
  **_user_id := gen_random_uuid();_**
  
  if you pass it as param, you can use the existing.

**Reply by GadMugambi** [reply]
I see.
My case will be a little difficult then. 
This is because the Kinde user id format in not a uuid but a text string. 
I don't know if Supabase allows that.

**Reply by mwolf1989** [reply]
dont think so:

<img width="355" alt="image" src="https://github.com/user-attachments/assets/669394a6-4f79-4802-8bcb-87b8786acaaf" />


this is the structure of the users table

**Reply by GadMugambi** [reply]
Oh. Thanks brother. 

I'll try to add the original Kinde user ids as metadata and still retain them in my tables.

Hopefully that works.

**Reply by mwolf1989** [reply]
or you add a profiles table, I use this mainly in the applications.

there I also store like stripe subscribe status etc.

**Reply by GadMugambi** [reply]
I think that will work. I'll try it.

I appreciate.

**Reply by mwolf1989** [reply]
Your welcome!
